cpp/src/arrow/util/benchmark_util.h

```diff
@@ -21,7 +21,10 @@
 
 #include "benchmark/benchmark.h"
 
+#include "arrow/memory_pool.h"
+#include "arrow/type_fwd.h"
 #include "arrow/util/cpu_info.h"
+#include "arrow/util/logging.h"  // IWYU pragma: keep
```

Comment:
We normally try to keep `logging.h` out of public headers.  Though I don't really know if `benchmark_util.h` is a public header or not :shrug: 

Why is everything defined inline here instead of in an cc file?

Reply:
I will put the implementation in a CC file.

Reply:
On second thought, this seems like more trouble than it's worth.